stratum/hal/lib/tai/module.cc

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "stratum/hal/lib/tai/module.h"
+
+#include <string>
+#include <vector>
+
+#include "stratum/hal/lib/tai/hostinterface.h"
+#include "stratum/hal/lib/tai/networkinterface.h"
+
+namespace stratum {
+namespace hal {
+namespace tai {
+
+Module::Module(const tai_api_method_table_t& api, const std::string& location)
+    : TAIObject(api), location_(location) {
+  LOG(INFO) << "Create Module with location: " << location;
+
+  std::vector<tai_attribute_t> list;
+  tai_attribute_t attr;
+  tai_status_t status;
+
+  attr.id = TAI_MODULE_ATTR_LOCATION;
+  attr.value.charlist.count = static_cast<uint>(location.size());
+  attr.value.charlist.list = const_cast<char*>(location_.c_str());
+  list.push_back(attr);
+
+  char str[] = "my vendor";
+  attr.id = TAI_MODULE_ATTR_VENDOR_NAME;
```

Comment:
As per Wataru's comments:
https://github.com/Telecominfraproject/oopt-tai/blob/master/inc/taimodule.h#L119
vendor name is a readonly attribute. You shouldn't set it when creating the module.
The stub library which you're using doesn't do the validation.